project_forecast_line_bokeh_chart/report/forecast_line_reporting.py

```diff
@@ -217,3 +220,8 @@ def _build_plots(self, height=300, width=1024):
             p.add_layout(p.legend[0], "right")
             plots.append(p)
         return plots
+
+    def _format_date(self, date):
+        if not date:
+            return ""
+        return datetime.datetime.strptime(date, "%Y-%m-%d").strftime("W%W %Y")
```

Comment:
Haha, yeah, this will solve it for week granularity :-) but for the month view for example it will be translated in the customer language, so it will be harder to replicate it like this.

Did you consider the version whereby we just display all the items in the dict in the right order, instead of querying for specific keys?

Reply:
My take on it was that it actually came into being because of a change you made - the backporting of the `_range` thing from 15.0 to 14.0. If we would make a slightly bigger change to not query for exact keys, it means we could scrap that backport, and actually end up with less custom code.

Reply:
@thomaspaulb please advise here, because I tried a couple of things and nothing works.

Reply:
My rough idea was that if we:

- make sure that [groupdata](https://github.com/sunflowerit/project/blob/a18350c1aedbe7f72185aebb8ff2ef869583f0e7/project_forecast_line_bokeh_chart/report/forecast_line_reporting.py#L99) comes out sorted by date

- change [this regular dict](https://github.com/sunflowerit/project/blob/a18350c1aedbe7f72185aebb8ff2ef869583f0e7/project_forecast_line_bokeh_chart/report/forecast_line_reporting.py#L130C45-L130C45) to be an `OrderedDict`

Then, [here](https://github.com/sunflowerit/project/blob/a18350c1aedbe7f72185aebb8ff2ef869583f0e7/project_forecast_line_bokeh_chart/report/forecast_line_reporting.py#L185) we can change:

```
                forecast = data[employee][project]
                plot_data[project] = [
                    forecast.get(self._format_date(date), 0) for date in dates
                ]
```

to

```
                forecast = data[employee][project]
                plot_data[project] = [f for f in forecast]
```

This should work unless `forecast` contains more dates than what gets placed in `dates` as returned from `_get_time_range()`; in that case, it will become more complicated. Still not impossible I think, but maybe first see if this will work.

Is this along the lines of what you tried? If not, what did you do?

Reply:
@thomaspaulb it took me a while but the solution was there, right in front of me. Adding an additional key in read_group() result solved everything.